tests/Service/PromotionsService/FindActivePromotionsByContextTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use DateTimeImmutable;
+
+class FindActivePromotionsByContextTest extends AbstractPromotionServiceTest
+{
+    /** @var CoreEntity */
+    private $context;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $dbId = 1;
+        $context = $this->mockEntity('Brand', $dbId);
+        $context->method('getDbAncestryIds')->willReturn([$dbId]);
+
+        $this->context = $context;
+    }
+
+    public function testFindActivePromotionsByEntityDefaultPagination()
+    {
+        $dbData = [['pid' => 'p000000h']];
+
+        $this->mockRepository
+            ->method('findActivePromotionsByContext')
+            ->with($this->context->getDbAncestryIds(), $this->isInstanceOf(DateTimeImmutable::class), 300, 0)
+            ->willReturn($dbData);
+
+        $promotions = $this->service()->findActivePromotionsByContext($this->context);
+
+        $this->assertEquals($this->getDomainModelsFromDbData($dbData), $promotions);
```

Comment:
Hi Michael, to be honest I think something similar to you. 

In this case we are mocking the db data, and the domain model Promotion is later not created with the mapper and mocked again. The only tests in here that make sense for me should be"I expect this method to be called, and this one as well". But compare data in here doesn't make sense. 

I did this following what we were making so far, and I think this should be improved in any way. Thinking about it...... Thanks for the comment, I thought was only me 😄